src/database/index.ts

```diff
@@ -14,33 +14,83 @@
+ * app or a given app.
+ *
+ * `admin.database()` can be called with no arguments to access the default
+ * app's {@link admin.database.Database `Database`} service or as
+ * `admin.database(app)` to access the
+ * {@link admin.database.Database `Database`} service associated with a specific
+ * app.
+ *
+ * `admin.database` is also a namespace that can be used to access global
+ * constants and methods associated with the `Database` service.
+ *
+ * @example
+ * ```javascript
+ * // Get the Database service for the default app
+ * var defaultDatabase = admin.database();
+ * ```
+ *
+ * @example
+ * ```javascript
+ * // Get the Database service for a specific app
+ * var otherDatabase = admin.database(app);
+ * ```
+ *
+ * @param App whose `Database` service to
+ *   return. If not provided, the default `Database` service will be returned.
+ *
+ * @return The default `Database` service if no app
+ *   is provided or the `Database` service associated with the provided app.
  */
+export declare function database(app?: app.App): database.Database;
+
+declare module '@firebase/database-types' {
+  interface FirebaseDatabase {
+    /**
+     * Gets the currently applied security rules as a string. The return value consists of
+     * the rules source including comments.
+     *
+     * @return A promise fulfilled with the rules as a raw string.
+     */
+    getRules(): Promise<string>;
+
+    /**
+     * Gets the currently applied security rules as a parsed JSON object. Any comments in
+     * the original source are stripped away.
+     *
+     * @return A promise fulfilled with the parsed rules object.
+     */
+    getRulesJSON(): Promise<object>;
+
+    /**
+     * Sets the specified rules on the Firebase Realtime Database instance. If the rules source is
+     * specified as a string or a Buffer, it may include comments.
+     *
+     * @param source Source of the rules to apply. Must not be `null` or empty.
+     * @return Resolves when the rules are set on the Realtime Database.
+     */
+    setRules(source: string | Buffer | object): Promise<void>;
+  }
+}
+
 /* eslint-disable @typescript-eslint/no-namespace */
-export namespace admin.database {
-  // See https://github.com/microsoft/TypeScript/issues/4336
+export namespace database {
   /* eslint-disable @typescript-eslint/no-unused-vars */
-  // See https://github.com/typescript-eslint/typescript-eslint/issues/363
-  export import DataSnapshot = firebaseDbTypesApi.DataSnapshot;
-  export import Database = adminDb.Database;
-  export import EventType = firebaseDbTypesApi.EventType;
-  export import OnDisconnect = firebaseDbTypesApi.OnDisconnect;
-  export import Query = firebaseDbTypesApi.Query;
-  export import Reference = firebaseDbTypesApi.Reference;
-  export import ThenableReference = firebaseDbTypesApi.ThenableReference;
-  export import enableLogging = firebaseDbTypesApi.enableLogging;
+  export import Database = rtdb.FirebaseDatabase;
```

Comment:
Can you directly export these types instead of nesting them in the namespace, so you don't have to do `import Database = database.Database;` everywhere?

Reply:
Our current API puts all RTDB types in an `admin.database` namespace. So this nesting is required.